murenn/dtcwt/nn.py

```diff
@@ -88,52 +86,66 @@ def to_conv1d(self):
         Compute the single-resolution equivalent impulse response of the MuReNN layer.
         This would be helpful for visualization in Fourier domain, for receptive fields,
         and for comparing computational costs.
-           DTCWT        conv1d        IDTCWT
-        δ -------> ψ_j --------> w_jq -------> y_jq
+           conv1d        IDTCWT
+        δ --------> w_jq -------> y_jq
         -------
         Return:
-            conv1d (torch.nn.Conv1d): A Pytorch Conv1d instance with weights initialized to y_jq.
+            conv1ds: A dictionary containing PyTorch Conv1d instances with weights initialized to y_jq.
+                - "complex" (torch.nn.Conv1d): the equivalent complex hybrid filter
+                - "real" (torch.nn.Conv1d): the real part of the hybrid filter
+                - "imag" (torch.nn.Conv1d): the imaginary part of the hybrid filter
         """
 
         device = self.conv1d[0].weight.data.device
-        # T the filter length
-        T = max(self.T)
-        # J the number of levels of decompostion
-        J = self.dtcwt.J
-        # Generate the impulse signal
-        N = 2 ** J * T
+        T = self.T  # Filter length
+        J = self.dtcwt.J  # Number of levels of decomposition
+        N = 2 ** J * max(T, len(self.dtcwt.g0a))  # Hybrid filter length
+
+        # Generate a zero signal
         x = torch.zeros(1, self.in_channels, N).to(device)
-        x[:, :, N//2] = 1
-        inv = murenn.IDTCWT(
-            J = J,
-            alternate_gh=False         
-        ).to(device)
-        # Get DTCWT impulse reponses
+
+        # Initialize the inverse DTCWT
+        inv = murenn.IDTCWT(J=J, alternate_gh=False).to(device)
```

Comment:
what's the status of alternate_gh now? are you sure it's still necessary to change the default to `False`? what happens if you set `alternate_gh=True`?

Reply:
Thanks for asking, this script shows the effect of `alternate_gh`:
```
from murenn import DTCWT, IDTCWT
import torch
from matplotlib import pyplot as plt
import numpy as np

def get_basis(j, **kwargs):
    N = 2**12
    x = torch.zeros(1, 1, N)

    fwd = DTCWT(**kwargs)
    inv = IDTCWT(**kwargs)

    lp, bpsr = fwd(x)
    _, bpsi = fwd(x)
    _, bpsc = fwd(x)


    bpsj = torch.zeros(bpsr[j].shape)
    bpsj[:,:,bpsj.shape[2]//2] = 1 / np.sqrt(2)**j

    bpsr[j] = bpsj * (1+0j)
    bpsi[j] = bpsj * (0+1j)


    yr = inv(lp, bpsr).detach()
    yi = inv(lp, bpsi).detach()

    return yr, yi

for j in range(8):
    yr, yi = get_basis(j, alternate_gh=False)
    yc = yr + 1j*yi
    yc_hat = np.fft.fft(yc.squeeze().numpy())
    plt.plot(np.abs(yc_hat)/np.abs(yc_hat).max())
plt.title("alternate_gh=False")

for j in range(8):
    yr, yi = get_basis(j, alternate_gh=True)
    yc = yr + 1j*yi
    yc_hat = np.fft.fft(yc.squeeze().numpy())
    plt.plot(np.abs(yc_hat)/np.abs(yc_hat).max())
plt.title("alternate_gh=True")
```

![image](https://github.com/user-attachments/assets/c4882cb9-dac3-4271-819a-d93037d1a279)
![image](https://github.com/user-attachments/assets/33789202-b395-4435-966b-d329bd423fae)